pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -503,6 +521,39 @@ class ToolDefinition:
     For MCP tools, this contains the `meta`, `annotations`, and `output_schema` fields from the tool definition.
     """
 
+    defer_loading: bool = False
+    """Whether to defer loading this tool until it's discovered via tool search.
+
+    When `True`, this tool will not be loaded into the model's context initially. Instead, Claude will
+    use the Tool Search tool to discover it on-demand when needed, reducing token usage.
+
+    Requires the `ToolSearchTool` builtin tool to be enabled.
+
+    Note: this is currently only supported by Anthropic models with the `advanced-tool-use-2025-11-20` beta.
+    See https://docs.anthropic.com/en/docs/agents-and-tools/tool-use/tool-search-tool for more info.
+    """
+
+    allowed_callers: list[str] | None = None
+    """List of tool types that are allowed to call this tool programmatically.
+
+    When set to `['code_execution_20250825']`, Claude can call this tool from within code execution,
+    enabling programmatic tool calling where Claude writes Python code that invokes your tools.
+
+    Note: this is currently only supported by Anthropic models with the `advanced-tool-use-2025-11-20` beta.
+    See https://docs.anthropic.com/en/docs/agents-and-tools/tool-use/code-execution-tool for more info.
+    """
+
+    input_examples: list[dict[str, Any]] | None = None
```

Comment:
Should this just be `examples` to be more generic? Or do we expect `output_examples` as well?